packages/block-library/src/cover/edit/index.js

```diff
@@ -114,25 +192,32 @@ function CoverEdit( {
 	);
 	const mediaUrl = media?.source_url;
 
-	useEffect( () => {
-		async function setIsDark() {
-			__unstableMarkNextChangeAsNotPersistent();
-			const isDarkSetting = await getCoverIsDark(
-				mediaUrl,
-				dimRatio,
-				overlayColor.color
-			);
-			setAttributes( {
-				isDark: isDarkSetting,
-			} );
-		}
-		if ( useFeaturedImage ) {
-			setIsDark();
-		}
-		// We only ever want to run this effect if the mediaUrl changes.
-		// All other changes to the isDark state are handled in the appropriate event handlers.
-		// eslint-disable-next-line react-hooks/exhaustive-deps
-	}, [ mediaUrl ] );
+	// set the is dark attribute based on the overlay
+	// and background color
+	const setIsDark = ( newOverlay, newBackground, newDimRatio = dimRatio ) => {
+		__unstableMarkNextChangeAsNotPersistent();
+		const isDarkSetting = computeIsDark(
+			newDimRatio,
+			newOverlay,
+			newBackground
+		);
+		setAttributes( {
+			isDark: isDarkSetting,
+		} );
+	};
+
+	// set the overlay color based on the average color
+	// of the image background
+	const setOverlayFromAverageColor = async ( newMedia ) => {
+		const newUrl =
+			newMedia?.url && newMedia.type === IMAGE_BACKGROUND_TYPE
+				? newMedia.url
+				: false;
+		__unstableMarkNextChangeAsNotPersistent();
+		const color = await getAverageBackgroundColor( newUrl );
+		setOverlayColor( color );
+		setIsDark( color, color );
```

Comment:
You don't have to run the composite source over operation on this since the colors are the same. `setAttributes( { isDark: colord( color ).isDark() } )` would work just as well.